### PR TITLE
[exporter/nopexporter] Add the nopexporter

### PR DESCRIPTION
**Pull request by evan-bradley**
**Description:**

Add the nopexporter. This can be helpful if a user wants to start the Collector with a dummy pipeline to only enable extensions. It could also be used to test Collector pipeline throughput.

**Link to tracking Issue:**

Resolves https://github.com/open-telemetry/opentelemetry-collector/issues/7316

**Testing:**

Added lifecycle tests; the receiver doesn't do anything.

**Documentation:**

Added a readme for the component.

cc @djaglowski @tigrannajaryan

### CODE REVIEW

**Review by astencel-sumo — APPROVED** [review]
:rocket: :tada:

**Comment by evan-bradley** [comment]
>My only bikeshed would be `nop` vs `noop` but both are acceptable spellings.

I would prefer `noop` myself, but it seems like `nop` is more conventional in Go and within the Collector. If others feel the same way, I'm open to changing the name.

**Comment by tigrannajaryan** [comment]
> > My only bikeshed would be `nop` vs `noop` but both are acceptable spellings.
> 
> I would prefer `noop` myself, but it seems like `nop` is more conventional in Go and within the Collector. If others feel the same way, I'm open to changing the name.

+1 to `nop`. One less character.

**Review by djaglowski — APPROVED** [review]

File: exporter/nopexporter/go.mod [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+go 1.21
+
+toolchain go1.21.6
+
+require (
+	github.com/stretchr/testify v1.8.4
+	go.opentelemetry.io/collector/component v0.94.1
+	go.opentelemetry.io/collector/consumer v0.94.1
+	go.opentelemetry.io/collector/exporter v0.93.0
+	go.opentelemetry.io/collector/pdata v1.1.0
```
Comment (dmitryax) [inline exporter/nopexporter/go.mod:12]: @evan-bradley can you please update the dependencies?

**Review by dmitryax — APPROVED** [review]

File: exporter/nopexporter/go.mod (hunk 1 above)
Comment (evan-bradley) [reply exporter/nopexporter/go.mod]: Thanks for the review, it should be good to go now. The failing check is only related to tests on generated code.

File: exporter/nopexporter/go.mod [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+go 1.21
+
+toolchain go1.21.6
+
+require (
+	github.com/stretchr/testify v1.8.4
+	go.opentelemetry.io/collector/component v0.95.0
+	go.opentelemetry.io/collector/consumer v0.95.0
+	go.opentelemetry.io/collector/exporter v0.93.0
```
[the remark this answers is not in the document]
Comment (evan-bradley) [reply exporter/nopexporter/go.mod]: Thanks for catching this. I've rebased and updated all versions to the latest release.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@evan-bradley please run `make gogenerate` to make CI green

**Comment by evan-bradley** [comment]
@dmitryax Took a bit of tweaking. Should be good to go now.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**